ShinyExpenseTrackerTests/src/ca/ualberta/cs/shinyexpensetracker/test/ExpenseItemActivityTest.java

```diff
@@ -0,0 +1,167 @@
+        	year = year;
+            month = monthOfYear;
+            day = dayOfMonth;
+        }
+    };
+
+
+    
+    protected void setUp() throws Exception {
+    	super.setUp();
+        instrumentation = getInstrumentation();
+        activity = (ExpenseItemActivity) getActivity();
+        
+        datePicker = new DatePickerDialog(instrumentation.getContext(), dateListener, TARGET_YEAR, TARGET_MONTH, TARGET_DAY);
+        
+        nameInput = ((EditText) activity.findViewById(ca.ualberta.cs.shinyexpensetracker.R.id.expenseItemNameEditText));
+        dateInput = ((EditText) activity.findViewById(ca.ualberta.cs.shinyexpensetracker.R.id.expenseItemDateEditText));
+        categoryInput = ((Spinner) activity.findViewById(ca.ualberta.cs.shinyexpensetracker.R.id.expenseItemCategorySpinner));
+        amountInput = ((EditText) activity.findViewById(ca.ualberta.cs.shinyexpensetracker.R.id.expenseItemAmountEditText));
+        currencyInput = ((Spinner) activity.findViewById(ca.ualberta.cs.shinyexpensetracker.R.id.expenseItemCurrencySpinner));
+        descriptionInput = ((EditText) activity.findViewById(ca.ualberta.cs.shinyexpensetracker.R.id.expesenItemDescriptionEditText));
+        photoInput = ((ImageButton) activity.findViewById(ca.ualberta.cs.shinyexpensetracker.R.id.expenseItemReceiptImageButton));
+        doneButton = (Button) activity.findViewById(ca.ualberta.cs.shinyexpensetracker.R.id.expenseItemDoneButton);
+    }
+    
+    /** Tests that when the ExpenseItemDateTextView is clicked a DatePickerDialog is shown */
+	public void testSetDateTimeField() {
+		instrumentation.runOnMainSync(new Runnable() {
+			public void run(){
+				dateInput.performClick();
+				
+				assertTrue("datepicker dialog is showing", ((ExpenseItemActivity) activity).getDialog().isShowing());
+			}
+		});
+	}
+	
+	/** test if an expense Item is successfully created */
+	public void testCreateExpenseItem() {
+		instrumentation.runOnMainSync(new Runnable(){
+			public void run() {
+				assertNotNull(doneButton);
+
+				 Calendar newDate = Calendar.getInstance();
+				 SimpleDateFormat sdf = new SimpleDateFormat();
+				 Date date = new Date();
+				 sdf.format(date);
+				 BigDecimal amount = new BigDecimal(10.00);
+				 Bitmap bitmap = null;
+				 
+				ExpenseItem expense = new ExpenseItem("name", date, 
+						 Category.ACCOMODATION,  amount, Currency.CAD, 
+						 "description", bitmap);
+
+				 assertTrue("name != name", "name" == expense.getName());
+				 assertTrue("date != name", "name" == expense.getName());
+				 assertTrue("category != accomodation", Category.ACCOMODATION == expense.getCategory());
+				 assertTrue("amount != 10.00", amount == expense.getAmountSpent());
+				 assertTrue("currnency != CAD", Currency.CAD == expense.getCurrency());
+				 assertTrue("description != description", "description" == expense.getDescription());
+				 assertTrue("bitmap != bitmap", bitmap == expense.getReceiptPhoto());
+			}
```

Comment:
* Can you change `assertTrue` to `assertEquals`? The "==" operator isn't reliable when comparing objects.
* Maybe also add one or two, things like: `assertNotSame( "something definitely not the value of expense.getName()", expense.getName() );` to make sure we aren't getting false positives.
* Can you add a test for `ExpenseItemActivity.createExpenseItem()`? ie. when I hit done, I want to be sure that everything I just entered into the UI is actually saved into the object.

Otherwise, looks good. :+1: